plugin/client.go

```diff
@@ -9,6 +9,8 @@ import (
 	"github.com/hashicorp/terraform/plugin/discovery"
 )
 
+var enableAutoMTLS = os.Getenv("TF_DISABLE_AUTOMTLS") == ""
```

Comment:
Could you add a comment attached to this stating that it's there only to support the plugin test framework, and not intended for any other use? Just thinking that a future maintainer finding this without any other context might be confused as to what it's for, and might be tempted to remove it.